lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -364,15 +176,81 @@ def processing_message
         say(msg)
       end
 
-      def run_rake
-        rake_success = ManageIQ::ApplianceConsole::Utilities.rake(task, task_params)
-        if rake_success && action == :restore && delete_agree
+      def run_action
+        success = send(action)
+        if success && action == :restore && delete_agree
           say("\nRemoving the database restore file #{uri}...")
           File.delete(uri)
-        elsif !rake_success
+        elsif !success
           say("\nDatabase #{action} failed. Check the logs for more information")
         end
       end
+
+      def backup
+        result = PostgresAdmin.backup(database_opts)
+        ManageIQ::ApplianceConsole.logger.info("[#{@database_opts[:dbname]}] database has been backed up to file: [#{uri}]")
+        result
+      end
+
+      def dump
+        result = PostgresAdmin.backup_pg_dump(database_opts)
+        ManageIQ::ApplianceConsole.logger.info("[#{@database_opts[:dbname]}] database has been dumped up to file: [#{uri}]")
+        result
+      end
+
+      def restore
+        backup_type = validate_backup_file_type
+
+        if application_connections?
+          message = "Database restore failed. Shut down all evmserverd processes before attempting a database restore"
+          ManageIQ::ApplianceConsole.logger.error(message)
+          raise message
+        end
+
+        MiqRegion.replication_type = :none
```

Comment:
@Fryguy and/or @jrafanie So most of the methods in "green" here are pulled from `lib/evm_database_ops.rb` in `ManageIQ/manageiq`:

https://github.com/ManageIQ/manageiq/blob/4abf8c6ddd9f1985c1264ca407837423695b1dfd/lib/evm_database_ops.rb#L183-L216

and I was able to convert most of what was there previously to raw `Pg` calls, but this one seems specific.  After taking a look myself, I wasn't able to determine why this line specifically was added, as it seems that it will always be effectively a "no-op" when the method involved is called:

https://github.com/ManageIQ/manageiq/blob/61c77793f44a2453c0908997d4a8c5f79607faa3/app/models/miq_region.rb#L156-L165

But something tells me I am missing something.  Can you enlighten me on what I am missing?

Reply:
`replication_type` describes whether or not this database is the global side of replication or the remote side of replication.  From a DB backup/restore point of view though I can't see why that would be important.

Reply:
Oh I think what's happening here is that it's temporarily disabling replication.  Setting replication_type to none would essentially unregister the replication provider.

Reply:
There was a bug where restore couldn't drop the database because the replication backend was holding a connection open I think. There could have very well been more to it than that, but it should all bit in git history/bugzilla somewhere.

Also :wave:

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1444993

PR: https://github.com/ManageIQ/manageiq/pull/16942

Reply:
@carbonin Okay, I think I see where my confusion was.

We pass into `.replication_type=` the `desired_type`, but the `if current_type` statements are getting the values from `.replication_type`, which will do some destroy/configure actions if needed, and I assume the methods called in `.replication_type` query against Postgres/PgLogical/stuffIVaguelyKnowAbout

Looks like I might need to look into what those do further, and see if we need to somehow replicate that logic over in the console.

Thanks for the response (and 👋 )

Reply:
oh hai @carbonin !! 👋